### PR TITLE
Blotter

### PR DESCRIPTION
**Pull request by MartinLavigneAdaptive**

### CODE REVIEW

File: src/e2e/src/pages/blotter.component.ts [hunk 1]
```diff
@@ -1,70 +1,53 @@
...
+    return trade
+  }
+
+  async getTradeDirection(num: string): Promise<ElementFinder> {
+    const trade = await this.root.element(by.qa(`${num}-direction`))
+    return trade
+  }
+
+  async getLatestTrade() {
+    const latestTrade = this.root.element(by.css("[col-id='tradeId']" + '.ag-cell'))
```
Comment (iowayankee) [inline src/e2e/src/pages/blotter.component.ts:48]: Nice!

File: src/e2e/src/pages/blotter.component.ts [hunk 2]
```diff
@@ -1,70 +1,98 @@
...
-          )
-        ),
-        tradeBackGroundColour: root.element(
-          by.css(
-            '[data-qa="shell-route__blotter-wrapper"] .ag-body .ag-row:nth-child(1) [col-id="statusIndicator"]'
-          )
-        )
-      }
-    }
+    // this.tradesTable = {
```
Comment (steinalex) [inline src/e2e/src/pages/blotter.component.ts:27]: Remove the commented code please :smile:

File: src/e2e/src/pages/blotter.component.ts [hunk 3]
```diff
@@ -1,70 +1,98 @@
...
+  }
+
+  async getTradeDealtCCY(num: string): Promise<ElementFinder> {
+    const trade = await this.root.element(by.qa(`${num}-dealtCurrency`))
+    return trade
+  }
+
+  async getTradeNotional(num: string): Promise<ElementFinder> {
+    const trade = await this.root.element(by.qa(`${num}-notional`))
+    return trade
```
Comment (steinalex) [inline src/e2e/src/pages/blotter.component.ts:79]: To make this even cleaner, you could just return the await without the `const` assignment.

`return await this.root.element(by.qa(`${num}-notional`))`

Same can be applied to the other functions

File: src/e2e/src/pages/blotter.component.ts [hunk 4]
```diff
@@ -1,70 +1,98 @@
...
 import { mapAsync } from '../utils/async.utils'
-import { waitForElementToBePresent } from '../utils/browser.utils'
+import { waitForElementToBePresent, waitForElementToBeVisible } from '../utils/browser.utils'
 
 export interface BlotterHeader {
   id: string
   label: string
 }
 
+type TradeProperties =
```
Comment (steinalex) [inline src/e2e/src/pages/blotter.component.ts:10]: If we're no longer using the constructor, these typings can be removed, as well as on line `24`

Comment (MartinLavigneAdaptive) [reply src/e2e/src/pages/blotter.component.ts:10]: This is actually used by a utils file, so this specific comment won't be addressed

**Pull request merged**